### PR TITLE
[#152966754] Cover RabbitMQ broker with unit tests

### PR DESCRIPTION
**Pull request by sergii-tkachenko**
#### What's this PR do?
- Full test coverage for RabbitMQ broker.

#### How should this be manually tested?
- `yarn install`
- `docker-compose up`
- `yarn test:full`

#### What are the relevant tickets?
Pivotal [#152966754](https://www.pivotaltracker.com/story/show/152966754)

### CODE REVIEW

File: src/lib/brokers/RabbitMQ/RabbitMQBroker.js [hunk 1]
```diff
@@ -25,13 +26,13 @@ class RabbitMQBroker extends Broker {
 
     // TODO: use options array instead of clientDescription.
     this.connectionManager = new RabbitMQConnectionManager(
-      amqpConfig.connection,
+      connection,
       clientDescription,
       reconnectManager,
     );
 
     // RabbitMQ exchange used for standart interfacing with queues.
```
Comment (rapala61) [inline src/lib/brokers/RabbitMQ/RabbitMQBroker.js]: Typo. 🕵️

File: test/unit/lib/brokers/RabbitMQ/RabbitMQBroker.test.js [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+});
+
+/**
+ * RabbitMQBroker: ack()
+ */
+test('RabbitMQBroker.ack(): Should delegate message ack to amqplib', (t) => {
+  // Set variables from the context.
+  const { sandbox, channel, broker } = t.context;
+
+  // Stub channel.sendImmediately() that sends all synchronous requests
```
Comment (rapala61) [inline test/unit/lib/brokers/RabbitMQ/RabbitMQBroker.test.js]: ... requests [to]? 🕵️

File: test/unit/lib/brokers/RabbitMQ/RabbitMQBroker.test.js [hunk 3]
```diff
@@ -0,0 +1,588 @@
...
+  // @see http://www.squaremobius.net/amqp.node/channel_api.html#channel_assertQueue
+  const assertQueueStub = sandbox.stub(channel, 'assertQueue').resolves({
+    queue: queueName,
+    messageCount: 0,
+    consumerCount: 0,
+  });
+
+  // Stub amqplib's bindQueue().
+  // It will return success for all calls except the third one.
+  // On the third call it'll sumulate unexpected error.
```
Comment (rapala61) [inline test/unit/lib/brokers/RabbitMQ/RabbitMQBroker.test.js]: Typo. 🕵️

File: test/unit/lib/brokers/RabbitMQ/RabbitMQBroker.test.js [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+    // `this` is amqplib's Channel.
+    this.registerConsumer(consumerTag, dequeuerSpy);
+    return consumerTag;
+  });
+
+  // Execute the function.
+  await broker.subscribe(queueName, dequeuerSpy);
+  consumeStub.should.have.been.calledOnce;
+
+  // Imitate broker senfing a message to the consumer.
```
Comment (rapala61) [inline test/unit/lib/brokers/RabbitMQ/RabbitMQBroker.test.js]: Typo. 🕵️

File: src/lib/brokers/RabbitMQ/RabbitMQBroker.js (hunk 1 above)
Comment (sergii-tkachenko) [reply src/lib/brokers/RabbitMQ/RabbitMQBroker.js]: Oh, what. I thought it's "starndart". https://english.stackexchange.com/questions/25423/why-do-non-native-english-users-often-spell-standard-as-standart

File: test/unit/lib/brokers/RabbitMQ/RabbitMQBroker.test.js (hunk 2 above)
Comment (sergii-tkachenko) [reply test/unit/lib/brokers/RabbitMQ/RabbitMQBroker.test.js]: Thanks, missed this on breaking the line.

File: src/lib/brokers/RabbitMQ/RabbitMQBroker.js (hunk 1 above)
Comment (sergii-tkachenko) [reply src/lib/brokers/RabbitMQ/RabbitMQBroker.js]: the more you know

**Pull request merged**